snapcraft/internal/repo.py

```diff
@@ -299,18 +319,13 @@ def _fix_symlinks(debdir):
         for entry in itertools.chain(files, dirs):
             path = os.path.join(root, entry)
             if os.path.islink(path) and os.path.isabs(os.readlink(path)):
-                target = os.path.join(debdir, os.readlink(path)[1:])
-                if _skip_link(os.readlink(path)):
-                    logger.debug('Skipping {}'.format(target))
-                    continue
-                if not os.path.exists(target):
-                    if not _try_copy_local(path, target):
-                        continue
-                os.remove(path)
-                os.symlink(os.path.relpath(target, root), path)
+                _fix_symlink(path, debdir, root)
             elif os.path.exists(path):
                 _fix_filemode(path)
 
+            if path.endswith('.pc') and not os.path.islink(path):
+                fix_pkg_config(debdir, path)
```

Comment:
Shouldn't you call fixup_func() rather here? (need to match the number of parameters) and move it to common as you are duplicating the checks for pc files + symlinks?